lib/orphans.nix

```diff
@@ -0,0 +1,43 @@
+{ lib }:
+let
+  inherit (lib) types optional isDerivation;
+  inherit (lib.lists) flatten any;
+  inherit (lib.asserts) assertMsg;
+  inherit (builtins) hasAttr length concatStringsSep mapAttrs attrValues tryEval;
```

Comment:
I would prefer if we would write out lib and use with builtins;

Reply:
Isn't a top-level `with` a bad thing?

Reply:
I think it's good the way it is — it makes it the uses of the functions concise without leaving it confusing where the functions are coming from.  Although all the functions you're importing from `lib.lists` and `lib.attrsets` are exported at the top level of lib, so you could reduce the lib imports to a single line.

Reply:
Nice catch @alyssais